sklearn/linear_model/tests/test_ridge.py

```diff
@@ -311,6 +312,123 @@ def test_ridge_individual_penalties():
+@pytest.mark.parametrize('y_shape', [(59,), (59, 1), (59, 3)])
+@pytest.mark.parametrize('noise', [1., 30.])
+def test_ridge_gcv_sample_weights(
+        gcv_mode, X_constructor, fit_intercept, n_features, y_shape, noise):
+    alphas = [1e-3, .1, 1., 10., 1e3]
+    rng = np.random.RandomState(0)
+    n_targets = y_shape[-1] if len(y_shape) == 2 else 1
+    x, y = datasets.make_regression(
+        n_samples=59, n_features=n_features, n_targets=n_targets,
+        random_state=0, shuffle=False, noise=noise)
+    y = y.reshape(y_shape)
+    x += 30 * rng.randn(x.shape[1])
+    sample_weight = 3 * rng.randn(len(x))
+    sample_weight = (sample_weight - sample_weight.min() + 1).astype(int)
+    indices = np.repeat(np.arange(x.shape[0]), sample_weight)
+    sample_weight = sample_weight.astype(float)
+    X_tiled, y_tiled = x[indices], y[indices]
+
+    cv = GroupKFold(n_splits=x.shape[0])
+    splits = cv.split(X_tiled, y_tiled, groups=indices)
+    kfold = RidgeCV(
+        alphas=alphas, cv=splits, scoring='neg_mean_squared_error',
+        fit_intercept=fit_intercept)
+    with ignore_warnings():
+        kfold.fit(X_tiled, y_tiled)
+
+    ridge_reg = Ridge(alpha=kfold.alpha_, fit_intercept=fit_intercept)
+    splits = cv.split(X_tiled, y_tiled, groups=indices)
+    predictions = cross_val_predict(ridge_reg, X_tiled, y_tiled, cv=splits)
+    kfold_errors = (y_tiled - predictions)**2
+    kfold_errors = [
+        np.sum(kfold_errors[indices == i], axis=0) for
+        i in np.arange(x.shape[0])]
+    kfold_errors = np.asarray(kfold_errors)
+
+    x_gcv = X_constructor(x)
+    gcv_ridge = RidgeCV(
+        alphas=alphas, store_cv_values=True,
+        gcv_mode=gcv_mode, fit_intercept=fit_intercept)
+    gcv_ridge.fit(x_gcv, y, sample_weight=sample_weight)
+    if len(y_shape) == 2:
+        gcv_errors = gcv_ridge.cv_values_[:, :, alphas.index(kfold.alpha_)]
+    else:
+        gcv_errors = gcv_ridge.cv_values_[:, alphas.index(kfold.alpha_)]
+
+    assert kfold.alpha_ == gcv_ridge.alpha_
+    assert_allclose(gcv_errors, kfold_errors, rtol=5e-2)
+    assert_allclose(gcv_errors, kfold_errors, rtol=5e-2)
+    assert_allclose(gcv_ridge.coef_, kfold.coef_, rtol=5e-2)
+    assert_allclose(gcv_ridge.intercept_, kfold.intercept_, rtol=5e-2)
+
+
+@pytest.mark.parametrize('mode', [True, 1, 5, 'bad', 'gcv', np.arange(3)])
+def test_check_gcv_mode_error(mode):
+    X, y = make_regression(n_samples=5, n_features=2)
+    gcv = RidgeCV(gcv_mode=mode)
+    with pytest.raises(ValueError, match="Unknown value for 'gcv_mode'"):
+        gcv.fit(X, y)
+    with pytest.raises(ValueError, match="Unknown value for 'gcv_mode'"):
+        _check_gcv_mode(X, mode)
+
+
+@pytest.mark.parametrize(
+    'mode, mode_samples_sup_features, mode_features_sup_samples',
+    [(None, 'svd', 'eigen'),
+     ('auto', 'svd', 'eigen'),
+     ('eigen', 'eigen', 'eigen'),
+     ('svd', 'svd', 'svd')]
+)
+def test_check_gcv_mode_choice(mode, mode_samples_sup_features,
```

Comment:
Could you also parameterize this test with:

```python
@pytest.mark.parametrize("sparse", [True, False])
@pytest.mark.parametrize(
    'mode, mode_samples_sup_features, mode_features_sup_samples',
    [(None, 'svd', 'eigen'),
     ('auto', 'svd', 'eigen'),
     ('eigen', 'eigen', 'eigen'),
     ('svd', 'svd', 'svd')]
)
def test_check_gcv_mode_choice(sparse, mode, mode_samples_sup_features,
```
```

And check that that the results holds when `X` is converted to a CSR matrix?

Reply:
Speaking of which, the docstring for `gcv_mode` needs to be updated to take the handling of sparse data into account. Currently it reads:

```
gcv_mode : {None, 'auto', 'svd', eigen'}, optional
    Flag indicating which strategy to use when performing
    Generalized Cross-Validation. Options are::

        'auto' : use svd if n_samples > n_features or when X is a sparse
                 matrix, otherwise use eigen
        'svd' : force computation via singular value decomposition of X
                (does not work for sparse matrices)
        'eigen' : force computation via eigendecomposition of X^T X

    The 'auto' mode is the default and is intended to pick the cheaper
    option of the two depending upon the shape and format of the training
    data.
```